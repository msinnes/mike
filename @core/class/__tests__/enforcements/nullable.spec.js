const nullableEnforcement = require('../../src/enforcements/nullable');

describe('nullableEnforcement', () => {
  it('should be a function', () => {
    expect(nullableEnforcement).toBeDefined();
    expect(nullableEnforcement).toBeInstanceOf(Function);
  });

  it('should not throw an error if input is a boolean or empty', () => {
    expect(() => {
      nullableEnforcement(true);
    }).not.toThrow();
    expect(() => {
      nullableEnforcement(false);
    }).not.toThrow();
    expect(() => {
      nullableEnforcement(null);
    }).not.toThrow();
    expect(() => {
      nullableEnforcement();
    }).not.toThrow();
  });

  it('should throw an error if an input is not a boolean', () => {
    expect(() => {
      nullableEnforcement([]);
    }).toThrowErrorMatchingSnapshot();
    expect(() => {
      nullableEnforcement({});
    }).toThrowErrorMatchingSnapshot();
  });
});
