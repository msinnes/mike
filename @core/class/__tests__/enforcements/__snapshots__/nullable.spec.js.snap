// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nullableEnforcement should throw an error if an input is not a boolean 1`] = `"InternalError: nullable must be a boolean"`;

exports[`nullableEnforcement should throw an error if an input is not a boolean 2`] = `"InternalError: nullable must be a boolean"`;
